upup/pkg/fi/cloudup/terraform/target_hcl2.go

```diff
@@ -87,7 +84,22 @@ func (t *TerraformTarget) finish012(taskMap map[string]fi.Task) error {
 
 	terraformBlock := rootBody.AppendNewBlock("terraform", []string{})
 	terraformBody := terraformBlock.Body()
-	terraformBody.SetAttributeValue("required_version", cty.StringVal(">= 0.12.0"))
+	terraformBody.SetAttributeValue("required_version", cty.StringVal(">= 0.12.26"))
+
+	requiredProvidersBlock := terraformBody.AppendNewBlock("required_providers", []string{})
+	requiredProvidersBody := requiredProvidersBlock.Body()
+
+	if t.Cloud.ProviderID() == kops.CloudProviderGCE {
+		requiredProviderBlockGoogle := requiredProvidersBody.AppendNewBlock("google = ", []string{})
+		requiredProviderBodyGoogle := requiredProviderBlockGoogle.Body()
+		requiredProviderBodyGoogle.SetAttributeValue("source", cty.StringVal("hashicorp/google"))
+		requiredProviderBodyGoogle.SetAttributeValue("version", cty.StringVal(">= 3.38.0"))
```

Comment:
I think rather than making the block name include an equals sign it might be more consistent to use the `writeMap` function in hcl2.go

(untested)
```suggestion
		writeMap(requiredProvidersBody, "google", map[string]cty.Value{
			"source": cty.StringVal("hashicorp/google"),
			"version": cty.StringVal(">= 3.38.0"),
		})
```

Reply:
I've tested the new block but it creates invalid Terraform HCL2 code as it encloses double quotes around both `source` and `version` keys. Is there a `cty` function to create this string as a key without the double quotes?

```
        terraform {
          required_version = ">= 0.12.26"
          required_providers {
            aws = {
              "source"  = "hashicorp/aws"
              "version" = ">= 3.6.0"
            }
          }
        }
```

Reply:
I think the double quotes are still valid, I tested it locally with 0.12 and 0.13 and init seemed to work fine:

```hcl
        terraform {
          required_version = ">= 0.12.26"
          required_providers {
            aws = {
              "source"  = "hashicorp/aws"
              "version" = ">= 3.6.0"
            }
          }
        }
```
```
providertest $ terraform0.12 init

Initializing the backend...

Initializing provider plugins...
- Checking for available provider plugins...
- Downloading plugin for provider "aws" (hashicorp/aws) 3.6.0...


Warning: Provider source not supported in Terraform v0.12

  on main.tf line 4, in terraform:
   4:             aws = {
   5:               "source"  = "hashicorp/aws"
   6:               "version" = ">= 3.6.0"
   7:             }

A source was declared for provider aws. Terraform v0.12 does not support the
provider source attribute. It will be ignored.

Terraform has been successfully initialized!

You may now begin working with Terraform. Try running "terraform plan" to see
any changes that are required for your infrastructure. All Terraform commands
should now work.

If you ever set or change modules or backend configuration for Terraform,
rerun this command to reinitialize your working directory. If you forget, other
commands will detect it and remind you to do so if necessary.
providertest $ terraform0.13 init

Initializing the backend...

Initializing provider plugins...
- Finding hashicorp/aws versions matching ">= 3.6.0"...
- Installing hashicorp/aws v3.6.0...
- Installed hashicorp/aws v3.6.0 (signed by HashiCorp)

Terraform has been successfully initialized!

You may now begin working with Terraform. Try running "terraform plan" to see
any changes that are required for your infrastructure. All Terraform commands
should now work.

If you ever set or change modules or backend configuration for Terraform,
rerun this command to reinitialize your working directory. If you forget, other
commands will detect it and remind you to do so if necessary.

```